wire-ios-data-model/Source/Core Crypto/CoreCryptoKeyProvider.swift

```diff
@@ -22,10 +22,18 @@ import WireSystem
 
 public class CoreCryptoKeyProvider {
 
-    public init() {}
+    private let coreCryptoKeyMigrationManager: CoreCryptoKeyMigrationManagerProtocol?
 
-    public func coreCryptoKey(createIfNeeded: Bool) throws -> Data {
+    public init(coreCryptoKeyMigrationManager: CoreCryptoKeyMigrationManagerProtocol?) {
+        self.coreCryptoKeyMigrationManager = coreCryptoKeyMigrationManager
+    }
+
+    public func coreCryptoKey(
+        createIfNeeded: Bool,
+        path: String
+    ) async throws -> Data {
         removeLegacyKeyIfNeeded()
+        try await migrateKeyIfNeeded(path: path)
```

Comment:
Aren't we missing marking the migration as not necessary when we don't have an old key we are end up creating a new one?

Reply:
Yes, good point. I implemented a fix.

CC throws an error: "SQL logic error".
Maybe it's better to improve the error description.